frontend/src/Components/RoadDetails/ImageGallery.tsx

```diff
@@ -72,8 +73,14 @@ const ImageGallery: React.FC = () => {
   const [scrollPosition, setScrollPosition] = useState(0);
   const galleryRef = useRef<HTMLDivElement>(null);
 
-  const openImageInNewTab = (imageUrl: string) => {
-    window.open(imageUrl, '_blank');
+  // Add feature for the pop-up of clicking and checking images
+  const [isImageClickOpen, setIsImageClickOpen] = useState(false);
+  const [currentImageIndex, setCurrentImageIndex] = useState(0);
```

Comment:
only the URL of the opened image is useful, so i would only store it

```suggestion
  const [openImageUrl, setOpenImageUrl] = useState("");
```